lntest/harness.go

```diff
@@ -846,6 +846,27 @@ func WaitPredicate(pred func() bool, timeout time.Duration) error {
 	}
 }
 
+// WaitInvariant is a helper test function that will wait for a timeout period
+// of time, verifying that a statement remains true for the entire duration.
+// This function is helpful as timing doesn't always line up well when running
+// integration tests with several running lnd nodes. This function gives callers
+// a way to assert that some property is maintained over a particular time
+// frame.
+func WaitInvariant(statement func() bool, timeout time.Duration) error {
+	exitTimer := time.After(timeout)
+	for {
+		select {
+		case <-exitTimer:
+			return nil
+		default:
+		}
+
+		if !statement() {
```

Comment:
This will be called rapidly in the current form. worth to have a small sleep at the end?

Reply:
Sure! 20ms?

Reply:
I would expect 2-5 seconds to be about as long as we'd need?

Reply:
sgtm ;)